build/xtask/src/dist.rs

```diff
@@ -1300,16 +1359,20 @@ pub fn allocate_all(
     let mut task_requests: BTreeMap<&str, BTreeMap<u32, VecDeque<&str>>> =
         BTreeMap::new();
 
-    for (name, task) in tasks {
-        for (mem, &amt) in &task.requires {
-            if !amt.is_power_of_two() {
-                bail!("task {}, memory region {}: requirement {} is not a power of two.",
-                    task.name, name, amt);
+    for name in tasks.keys() {
+        for (mem, amt) in task_sizes[name.as_str()].iter() {
+            let bytes = toml.suggest_memory_region_size(name, *amt);
+            if let Some(r) = tasks[name].max_sizes.get(&mem.to_string()) {
+                if bytes > *r as u64 {
+                    bail!(
+                        "task {}, memory region {}: requirement {} is too small (needs {}).",
+                        name, name, r, bytes);
```

Comment:
This is wrong, for the current format string it should be `name, mem, r, bytes`

Reply:
Yup, fixed